avocado/core/test.py

```diff
@@ -313,11 +318,13 @@ def _get_tagged_name(self, logdir):
             name += ".%s" % self.tag
         tag = 0
         tagged_name = name
-        while os.path.isdir(os.path.join(logdir, tagged_name)):
+        safe_tagged_name = astring.string_to_safe_path(tagged_name)
+        while os.path.isdir(os.path.join(logdir, safe_tagged_name)):
             tag += 1
             tagged_name = "%s.%s" % (name, tag)
-        self.tag = "%s.%s" % (self.tag, tag) if self.tag else str(tag)
+            safe_tagged_name = astring.string_to_safe_path(tagged_name)
 
+        self.tag = "%s.%s" % (self.tag, tag) if self.tag else str(tag)
         return tagged_name
```

Comment:
Why not to `return astring.string_to_safe_path(tagged_name)` and avoid line 107?